invokeai/frontend/web/src/features/nodes/components/flow/nodes/Invocation/fields/FieldLinearViewToggle.tsx

```diff
@@ -0,0 +1,74 @@
+import { IconButton } from '@invoke-ai/ui-library';
+import { createSelector } from '@reduxjs/toolkit';
+import { useAppDispatch, useAppSelector } from 'app/store/storeHooks';
+import {
+  selectWorkflowSlice,
+  workflowExposedFieldAdded,
+  workflowExposedFieldRemoved,
+} from 'features/nodes/store/workflowSlice';
+import { memo, useCallback, useMemo } from 'react';
+import { useTranslation } from 'react-i18next';
+import { PiMinusBold, PiPlusBold } from 'react-icons/pi';
+
+type Props = {
+  nodeId: string;
+  fieldName: string;
+  isHovered: boolean;
+};
+
+const FieldLinearViewToggle = ({ nodeId, fieldName, isHovered }: Props) => {
+  const dispatch = useAppDispatch();
+  const { t } = useTranslation();
+
+  const selectIsExposed = useMemo(
+    () =>
+      createSelector(selectWorkflowSlice, (workflow) => {
+        return Boolean(workflow.exposedFields.find((f) => f.nodeId === nodeId && f.fieldName === fieldName));
+      }),
+    [fieldName, nodeId]
+  );
+
+  const isExposed = useAppSelector(selectIsExposed);
+
+  const handleExposeField = useCallback(() => {
+    dispatch(workflowExposedFieldAdded({ nodeId, fieldName }));
+  }, [dispatch, fieldName, nodeId]);
+
+  const handleUnexposeField = useCallback(() => {
+    dispatch(workflowExposedFieldRemoved({ nodeId, fieldName }));
+  }, [dispatch, fieldName, nodeId]);
+
+  const ToggleButton = useMemo(() => {
```

Comment:
I appreciate the memoizing but it's not needed for this component - we can just return directly:

```ts
if (!isHovered) {
  return null
}

if (!isExposed) {
  return ...
}

if (isExposed) {
  return ...
}
```